source/common/config/grpc_mux_impl.cc

```diff
@@ -206,12 +210,14 @@ void GrpcMuxImpl::onDiscoveryResponse(
 void GrpcMuxImpl::onWriteable() { drainRequests(); }
 
 void GrpcMuxImpl::onStreamEstablished() {
+  first_stream_request_ = true;
   for (const auto& type_url : subscriptions_) {
     queueDiscoveryRequest(type_url);
   }
 }
 
 void GrpcMuxImpl::onEstablishmentFailure() {
+  first_stream_request_ = true;
```

Comment:
Is this strictly needed? Won't the `onStreamEstablished` cover this case when the connection is re-established?

Reply:
You are right, it's not strictly necessary since no message is sent until stream is established.